atd-vzv/src/views/summary/PeopleByDemographics.js

```diff
@@ -1,258 +1,284 @@
+      colors.demoAge6of8,
+      colors.demoAge7of8,
+      colors.demoAge8of8,
+    ],
+  },
+  genderData: {
+    label: "Gender",
+    categoryKey: "prsn_gndr_id",
+    categoryType: "target",
+    socrataCategories: ["1", "2", "0"],
+    labelCategories: ["Male", "Female", "Unknown"],
+    exceptionCategory: "Unknown",
+    query: `SELECT date_extract_y(crash_date) as year, COUNT(*) as total, prsn_gndr_id, prsn_injry_sev_id
+              WHERE ${dateCondition}
+              GROUP BY year, prsn_gndr_id, prsn_injry_sev_id
+              ORDER BY year`,
+    colors: chartColors,
+  },
+};
+
+const PeopleByDemographics = () => {
   const [chartType, setChartType] = useState("Race/Ethnicity");
-  const [chartData, setChartData] = useState(null); // {yearInt: [{record}, {record}, ...]}
-  const [crashType, setCrashType] = useState([]);
+  const [crashType, setCrashType] = useState(null);
+  const [chartData, setChartData] = useState({});
+  const [demoData, dispatch] = useReducer(demoDataReducer, {});
 
-  // Fetch data and set in state by years in yearsArray
+  function demoDataReducer(demoData, action) {
+    switch (action.type) {
+      case "setData":
+        return { ...demoData, ...action.payload };
+      default:
+        return null;
+    }
+  }
+
+  // Fetch data for each visualization in config
   useEffect(() => {
-    // Wait for crashType to be passed up from setCrashType component
-    if (crashType.queryStringPerson) {
-      const getChartData = async () => {
-        let newData = {};
-        // Use Promise.all to let all requests resolve before setting chart data by year
-        await Promise.all(
-          yearsArray().map(async (year) => {
-            // If getting data for current year (only including years past January),
-            // set end of query to last day of previous month,
-            // else if getting data for previous years, set end of query to last day of year
-            let endDate =
-              year.toString() === dataEndDate.format("YYYY")
-                ? `${dataEndDate.format("YYYY-MM-DD")}T23:59:59`
-                : `${year}-12-31T23:59:59`;
-            let url = `${personEndpointUrl}?$where=${crashType.queryStringPerson} AND crash_date between '${year}-01-01T00:00:00' and '${endDate}'`;
-            await axios.get(url).then((res) => {
-              newData = { ...newData, ...{ [year]: res.data } };
-            });
-            return null;
-          })
+    if (crashType !== null && Object.keys(demoData).length === 0) {
+      const requests = Object.entries(chartConfigs).map(([dataKey, config]) =>
+        axios.get(url + encodeURIComponent(config.query), {
+          dataKey,
+        })
+      );
+
+      axios.all(requests).then((resArr) => {
+        const responses = resArr.reduce(
+          (acc, res) => ({ ...acc, ...{ [res.config.dataKey]: res.data } }),
+          {}
         );
-        setChartData(newData);
-      };
-      getChartData();
+
+        dispatch({
```

Comment:
i'm not sure what is the proper React convention, but just thinking it might make sense to call this `demoDataDispatch` or something to differentiate if you used `useReducer` for something else in  this component. Just a thought more than a suggestion since `useReducer` is still new for me.

Reply:
Nice thinking - I keep fighting an impulse to think of the state in `useReducer` as a catch all for state in a component for some reason. Good to have more thoughts pushing me away from that and thinking about keeping the code easy to read when state gets even more complex. Things could get buggy real fast without setting things apart in this way. 🐛